src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
+const INITIAL_HEALTH = 100;
+const BITE_DAMAGE = 50;
+const INITIAL_HIDDEN_STATE = false;
+
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = INITIAL_HEALTH) {
+    this.name = name;
+    this.health = health;
+
+    Animal.addAnimal(this);
+  }
+
+  static addAnimal(animal) {
+    Animal.alive.push(animal);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = INITIAL_HEALTH, hidden = INITIAL_HIDDEN_STATE) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
```

Comment:
The 'hide' method currently sets the 'hidden' property to true, which is correct according to the checklist. However, ensure that there is no requirement for a method to toggle the 'hidden' state, as the current implementation does not allow for 'unhiding'. If a toggle or 'unhide' functionality is needed, it should be implemented as a separate method.